plugins/builds/update.js

```diff
@@ -245,18 +268,58 @@ module.exports = () => ({
 
             const skipFurther = /\[(skip further)\]/.test(newEvent.causeMessage);
 
-            // Guard against triggering non-successful or unstable builds
-            // Don't further trigger pipeline if intented to skip further jobs
+            // Update stageBuild status if it has changed;
+            // if stageBuild status is currently terminal, do not update
+            const stage = await getStage({
+                stageFactory,
+                workflowGraph: newEvent.workflowGraph,
+                jobName: job.name,
+                pipelineId: pipeline.id
+            });
+            const isStageTeardown = STAGE_TEARDOWN_PATTERN.test(job.name);
+            let stageBuildHasFailure = false;
+
+            if (stage) {
+                const stageBuild = await stageBuildFactory.get({
+                    stageId: stage.id,
+                    eventId: newEvent.id
+                });
+
+                if (stageBuild && stageBuild.status !== newBuild.status) {
+                    if (!['FAILURE', 'ABORTED', 'UNSTABLE'].includes(stageBuild.status)) {
+                        stageBuild.status = newBuild.status;
+                        await stageBuild.update();
+                    }
+                }
+
+                stageBuildHasFailure = ['FAILURE', 'ABORTED', 'UNSTABLE'].includes(stageBuild.status);
+            }
 
+            // Guard against triggering non-successful or unstable builds
+            // Don't further trigger pipeline if intend to skip further jobs
             if (newBuild.status !== 'SUCCESS' || skipFurther) {
                 // Check for failed jobs and remove any child jobs in created state
                 if (newBuild.status === 'FAILURE') {
-                    await removeJoinBuilds({ pipeline, job, build: newBuild }, request.server.app);
+                    await removeJoinBuilds(
```

Comment:
`removeJoinBuild()` needed to be invoked when the current build is `teardown` and has successfully completed, but the `stageBuild` status is a negative terminal state